syntax = "proto3";

message NoUserMessage{}

message ID {
    int32 id = 1;
}

message User {
    int32 id = 1;
    string name = 2;
    int32 age = 3;
    string password = 4;
    optional string email = 5;
}

message UserList {
    repeated User users = 1;
}

service UserService{
    rpc CreateUser (User) returns (User);
    rpc GetUser (ID) returns (User);
    rpc GetUserList (NoUserMessage) returns (UserList);
    rpc UpdateUser (User) returns (User);
    rpc DeleteUser (ID) returns (User);
}